patches/tModLoader/Terraria/UI/ItemSlot.cs.patch

```diff
@@ -260,40 +412,145 @@
  			}
  
  			for (int i = 0; i < itemCollection.Length; i++) {
-@@ -1917,7 +_,7 @@
+-				if (slot < 10 && i < 10) {
+-					if (item.wingSlot > 0 && itemCollection[i].wingSlot > 0)
++				if (item.wingSlot > 0 && itemCollection[i].wingSlot > 0)
+-						return true;
++					return true;			
++			}
+ 
+-					if (slot >= 10 && i >= 10 && item.wingSlot > 0 && itemCollection[i].wingSlot > 0)
+-						return true;
+-				}
+-
++			for (int i = 0; i < itemCollection.Length; i++) {
+ 				if (item.IsTheSameAs(itemCollection[i]))
  					return true;
  			}
  
 -			return false;
 +			return !ItemLoader.CanEquipAccessory(item, slot);
  		}
  
++		//TML: I (Solxan), have not been able to get this to ever run in-game. I suspect this code path is deprecated.
  		private static Item DyeSwap(Item item, out bool success) {
-@@ -1993,6 +_,14 @@
- 						accSlotToSwapTo = j - 3;
+ 			success = false;
+ 			if (item.dye <= 0)
+@@ -1934,9 +_,14 @@
  				}
+ 			}
  
-+				for (int k = 0; k < num2; k++) {
-+					int index = 3 + (accSlotToSwapTo + num2) % num2;
-+					if (ItemLoader.CanEquipAccessory(item, index)) {
-+						accSlotToSwapTo = index - 3;
-+						break;
-+					}
-+				}
+-			if (dyeSlotCount >= 10)
++			if (dyeSlotCount >= 10) {
++				item2 = LoaderManager.Get<AccessorySlotLoader>().VanillaDyeSwapMirror(item, out success);
++				if (success)
++					return item2;
 +
- 				if (accSlotToSwapTo > num2)
- 					return item;
+ 				dyeSlotCount = 0;
+-
++			}
++			
+ 			if (dyeSlotCount < 0)
+ 				dyeSlotCount = 9;
  
-@@ -2008,6 +_,9 @@
- 						num3 = k;
+@@ -1974,43 +_,70 @@
+ 				player.armor[num + 2] = item.Clone();
+ 			}
+ 			else if (item.accessory) {
++				//TML: Rewrote ArmorSwap for accessories under the PR #1299 so it was actually readable. No vanilla functionality lost in transition
```

Comment:
preferred approach to a whole method rewrite is to make another method and call it while commenting out the vanilla impl